code/modules/mob/living/carbon/xenomorph/castes/king/abilities_king.dm

```diff
@@ -485,16 +485,22 @@
 	xeno_message("King: \The [owner] has begun a psychic summon in <b>[get_area(owner)]</b>!", hivenumber = X.hivenumber)
 	var/list/allxenos = X.hive.get_all_xenos()
 	for(var/mob/living/carbon/xenomorph/sister AS in allxenos)
+		if(sister.z != owner.z)
+			continue
 		sister.add_filter("summonoutline", 2, outline_filter(1, COLOR_VIOLET))
 
 	if(!do_after(X, 10 SECONDS, FALSE, X, BUSY_ICON_HOSTILE))
```

Comment:
```suggestion
	if(!do_after(X, 10 SECONDS, FALSE, X, BUSY_ICON_HOSTILE))
```
Like I'd add a callback or something to this.

Reply:
why would you want the teleport to cancel if one random xeno changes z-level?

Reply:
If **you** change Z level

Reply:
That's not what this is checking tho